### PR TITLE
debug/failed multi

### PR DESCRIPTION
**Pull request by NgoKimPhu**
- **debug: add stack trace for panic in CalcAmountOut**
- **fix: panic in woofi21 and dexalot and clipper**
- **fix: dexalot logic unexpectedly modifying price levels; refactor**
- **fix: curve stable-plain underflow when used as base pool (EX-2354)**

### CODE REVIEW

File: pkg/liquidity-source/woofi-v21/pool_simulator.go [hunk 1]
```diff
@@ -318,7 +320,10 @@ func (s *PoolSimulator) _calcBaseAmountSellQuote(
 
 	decs := s.decimalInfo(baseToken)
 
-	if quoteAmount.Cmp(s.tokenInfos[baseToken].MaxNotionalSwap) > 0 {
+	if maxNotionalSwap := s.tokenInfos[baseToken].MaxNotionalSwap; maxNotionalSwap == nil {
+		logger.Warnf("no MaxNotionalSwap for %s in %v: %v", baseToken, s.tokenInfos, s.tokenInfos[baseToken])
```
Comment (NgoKimPhu) [inline pkg/liquidity-source/woofi-v21/pool_simulator.go:324]: why is this nil?

**Review by lehainam-dev — APPROVED** [review]
Approve for Clipper's pool_simulator & update in `pkg/source/pool/pool.go`

**Review by sunspirit99 — APPROVED** [review]
a good patch :pray:

**Review by tien7668 — APPROVED** [review]

File: pkg/liquidity-source/woofi-v21/pool_simulator.go (hunk 1 above)
Comment (NgoKimPhu) [reply pkg/liquidity-source/woofi-v21/pool_simulator.go]: fixed, due to UpdateBalance

**Pull request merged**